### ISSUE TITLE
`nrNsaBandsEutra` and `nrSaBandsEutra` are instances of BandNrDetails but only contain band numbers

### ISSUE THREAD

**Issue by davwheat** [HandyMenny/uecapabilityparser#234]
I'm currently implementing NR band details into my parse/import API.

Looking through some examples I have, the tests in this repo, and the source, it seems `nrNsaBandsEutra` and `nrSaBandsEutra` fields are both instances of `List<BandNrDetails>` but the source will only ever set the band attribute to a value:

https://github.com/HandyMenny/uecapabilityparser/blob/f198922a1001b4658dd30a9ca19fb9084b390a7e/src/main/java/it/smartphonecombo/uecapabilityparser/importer/ImportCapabilityInformation.kt#L675-L678

This makes the TS typings a bit misleading, and would have significantly increased complexity if I hadn't noticed this.

Could the output be simplified to account for this? Or is this preparing for a future version where this isn't the case?

**Comment by HandyMenny** [comment]
> Could the output be simplified to account for this?

Yes, indeed. But I would like to leave the current boxing (e.g. {band: X}), to avoid a breaking change



> Or is this preparing for a future version where this isn't the case?

Unlikely, the current standard has only band numbers
___
There's a similar situation with filters:

https://github.com/HandyMenny/uecapabilityparser/blob/f198922a1001b4658dd30a9ca19fb9084b390a7e/src/main/java/it/smartphonecombo/uecapabilityparser/model/filter/UeCapabilityFilterLte.kt#L11-L13

https://github.com/HandyMenny/uecapabilityparser/blob/f198922a1001b4658dd30a9ca19fb9084b390a7e/src/main/java/it/smartphonecombo/uecapabilityparser/model/filter/UeCapabilityFilterNr.kt#L10-L12

https://github.com/HandyMenny/uecapabilityparser/blob/f198922a1001b4658dd30a9ca19fb9084b390a7e/src/main/java/it/smartphonecombo/uecapabilityparser/model/filter/BandFilterLte.kt#L9-L13

`lteBands` in `UeCapabilityFilterLte` contains only bands, while `lteBands` in `UeCapabilityFilterNr` can also contain DL and UL class. Do you think it should be simplified here as well?

**Comment by davwheat** [comment]
IMO, simplify whatever can be simplified 😅

If the output can never be generated, there's no need for it to be present in the typings, and only makes implementing an import system more difficult.

Leaving the boxing makes sense to me since it allows for future additions if new spec versions allow for more data, for example, and that wouldn't need to be a breaking change. 

I'll take a look at the filters you listed in a moment because I don't trust GitHub mobile to keep my comment draft...

**Comment by davwheat** [comment]
> lteBands in UeCapabilityFilterLte contains only bands, while lteBands in UeCapabilityFilterNr can also contain DL and UL class. Do you think it should be simplified here as well?

Yeah, I don't see why not. It makes sense to remove what isn't used here, too.

I'm not sure whether the better approach is (I haven't looked at how the typings are generated) some custom interface that only implements the few fields needed but reuse the same model? I don't know if that would work or not.

**Comment by HandyMenny** [comment]
> I haven't looked at how the typings are generated

Typescript types are generated by [Kotlinx Serialization TypeScript Generator](https://github.com/adamko-dev/kotlinx-serialization-typescript-generator), which uses the same source (kotlinx-serialization) used for JSON serialization/de-serialization
